### PR TITLE
Fix `createLabel` option should not be clickable for `<AutocompleteInput>` and `<AutocompleteArrayInput>`

### PR DESCRIPTION
**Pull request by slax57**
## Problem

The [documentation](https://marmelab.com/react-admin/AutocompleteInput.html#createlabel) encourages to use values like the following for `createLabel`:
```tsx
<AutocompleteInput
    source="author"
    choices={authors}
    onCreate={onCreate}
    createLabel="Start typing to create a new item"
/>
```
One would expect this label to be only a hint, but this `createLabel` option is actually clickable, and what's more, it pre-selects `"Start typing to create a new item"` as value.

It would seem more logical to [disable](https://mui.com/material-ui/react-autocomplete/#disabled-options) this option, to prevent the bug and make it clear that it's just a hint.

## Solution

Upon further analysis, having a clickable `createLabel` option actually makes sense for components that do not allow to type a value in, like `<SelectInput>` and `<SelectArrayInput>`.

But for components like `<AutocompleteInput>` and `<AutocompleteArrayInput>`, we cannot know of the `createLabel` option is supposed to be clickable or not, because it depends on the intent conveyed by the actual chosen label (`createLabel="Start typing to create a new item"` should not be clickable, but `createLabel="Click to create a new item"` should be).

Since the `<AutocompleteInput>` and `<AutocompleteArrayInput>` also support a `createItemLabel` option, we decided that this would be the only way to start the creation workflow for these components, and hence the `createLabel` option will always be considered as a hint (and be disabled).

TLDR: Components supporting both `createItemLabel` and `createLabel` will consider `createLabel` as a hint (option will be disabled), while components supporting only `createLabel` will consider this option clickable.

## Todo

- [x] Update `useSupportCreateSuggestion` to work as described
- [x] Add stories
- [x] Add tests
- [x] Update documentation

## How To Test

- Storybooks
  - AutocompleteArrayInput
  - AutocompleteInput
  - SelectArrayInput
  - SelectInput
- Run the unit tests

## Additional Checks

- [x] The PR targets `master` for a bugfix, or `next` for a feature
- [x] The PR includes **unit tests** (if not possible, describe why)
- [x] The PR includes one or several **stories** (if not possible, describe why)
- [x] The **documentation** is up to date

Also, please make sure to read the [contributing guidelines](https://github.com/marmelab/react-admin#contributing).

### CODE REVIEW

File: packages/ra-ui-materialui/src/input/AutocompleteInput.stories.tsx [hunk 1]
```diff
@@ -504,6 +504,36 @@ export const CreateItemLabel = () => (
...
+                setChoices(options => [...options, newOption]);
+                // Wait until next tick to give some time for React to update the state
+                await new Promise(resolve => setTimeout(resolve));
+                return newOption;
+            }}
+            createItemLabel="" // TODO - FIX THIS
+            createLabel="Click to create a new item"
+        />
+    );
+};
```
Comment (slax57) [inline packages/ra-ui-materialui/src/input/AutocompleteInput.stories.tsx:529]: I believe this use case would be nice for people still wanting to start the creation workflow with the `createLabel` and not the `createItemLabel`. All they would need to do would be to pass `createItemLabel=""`.
So I'll see if I can make it work (as it currently doesn't).

File: packages/ra-ui-materialui/src/input/AutocompleteInput.tsx [hunk 2]
```diff
@@ -522,7 +536,11 @@ If you provided a React element for the optionText prop, you must also provide t
             if (inputValue === '' && filterValue === '' && createLabel) {
                 // create option with createLabel
                 filteredOptions = filteredOptions.concat(getCreateItem(''));
-            } else if (!doesQueryMatchSuggestion(filterValue)) {
+            } else if (
+                inputValue &&
+                filterValue &&
+                !doesQueryMatchSuggestion(filterValue)
+            ) {
```
Comment (slax57) [inline packages/ra-ui-materialui/src/input/AutocompleteInput.tsx:543]: This is actually not directly linked to the main fix, but fixes an issue where `<AutocompleteArrayInput>` would still render the 'Create' option when the filter is empty (i.e. the `createLabel` option), although this should no longer be the case according to this [PR](https://github.com/marmelab/react-admin/pull/10266).

File: packages/ra-ui-materialui/src/input/useSupportCreateSuggestion.tsx [hunk 3]
```diff
@@ -19,24 +19,28 @@ import set from 'lodash/set';
...
+ * - getOptionDisabled: a function which should be passed to the input to disable the create choice when the filter is empty (to make it a hint).
  */
 export const useSupportCreateSuggestion = (
     options: SupportCreateSuggestionOptions
 ): UseSupportCreateValue => {
     const {
         create,
         createLabel = 'ra.action.create',
-        createItemLabel = 'ra.action.create_item',
+        createItemLabel,
```
Comment (slax57) [inline packages/ra-ui-materialui/src/input/useSupportCreateSuggestion.tsx:41]: One could argue this is technically a BC, but I thought it was worth it to reuse the `createItemLabel` prop to decide whether or not to make the `createLabel` option clickable, rather than introducing yet another prop.
Besides, this hook can probably be considered an internal hook, as it is not documented.

File: packages/ra-ui-materialui/src/input/AutocompleteArrayInput.spec.tsx [hunk 4]
```diff
@@ -905,7 +951,7 @@ describe('<AutocompleteArrayInput />', () => {
                 selector: 'input',
             })
         );
-        expect(screen.queryAllByRole('option')).toHaveLength(3);
+        expect(screen.queryAllByRole('option')).toHaveLength(2);
```
Comment (slax57) [inline packages/ra-ui-materialui/src/input/AutocompleteArrayInput.spec.tsx:954]: This is because we no longer display the 'Create' item when the input is empty.

File: docs/AutocompleteArrayInput.md [hunk 5]
```diff
@@ -60,8 +60,8 @@ The form value for the source must be an array of the selected values, e.g.
...
+| `createLabel`              | Optional | `string`              | -                        | The label used as hint to let users know they can create a new choice. Displayed when the filter is empty.                                       |
+| `createItemLabel`          | Optional | `string`              | `ra.action .create_item` | The label for the menu item allowing users to create a new choice. Used when the filter is not empty.                                                    |
 | `debounce`                 | Optional | `number`              | `250`                    | The delay to wait before calling the setFilter function injected when used in a ReferenceArray Input.                                                         |
 | `emptyValue`               | Optional | `any`                 | `''`                     | The value to use for the empty element                                                                                                                  |
```
Comment (erwanMarmelab) [inline docs/AutocompleteArrayInput.md:66]:
```suggestion
| `emptyValue`               | Optional | `any`                 | `''`                     | The value to use for the empty element.                                                                                                                 |
```

File: docs/AutocompleteArrayInput.md [hunk 6]
```diff
@@ -60,8 +60,8 @@ The form value for the source must be an array of the selected values, e.g.
 |----------------------------|----------|-----------------------|--------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
 | `choices`                  | Required | `Object[]`            | -                        | List of choices                                                   |
 | `create`                   | Optional | `Element`             | `-`                      | A React Element to render when users want to create a new choice                                                                       |
```
Comment (erwanMarmelab) [inline docs/AutocompleteArrayInput.md:62]:
```suggestion
| `create`                   | Optional | `Element`             | `-`                      | A React Element to render when users want to create a new choice.                                                                      |
```

File: docs/AutocompleteArrayInput.md [hunk 7]
```diff
@@ -60,8 +60,8 @@ The form value for the source must be an array of the selected values, e.g.
...
+| `createItemLabel`          | Optional | `string`              | `ra.action .create_item` | The label for the menu item allowing users to create a new choice. Used when the filter is not empty.                                                    |
 | `debounce`                 | Optional | `number`              | `250`                    | The delay to wait before calling the setFilter function injected when used in a ReferenceArray Input.                                                         |
 | `emptyValue`               | Optional | `any`                 | `''`                     | The value to use for the empty element                                                                                                                  |
 | `filterToQuery`            | Optional | `string` => `Object`  | `q => ({ q })`           | How to transform the searchText into a parameter for the data provider                                                                                  |
```
Comment (erwanMarmelab) [inline docs/AutocompleteArrayInput.md:67]:
```suggestion
| `filterToQuery`            | Optional | `string` => `Object`  | `q => ({ q })`           | How to transform the searchText into a parameter for the data provider.                                                                                 |
```

same for the following props (but GH doesn't allow me to add suggestions there)

File: docs/SelectArrayInput.md [hunk 8]
```diff
@@ -223,6 +223,21 @@ If you just need to ask users for a single string to create the new option, you
 
 If you're in a `<ReferenceArrayInput>` or `<ReferenceManyToManyInput>`, the `handleSubmit` will need to create a new record in the related resource. Check the [Creating New Choices](#creating-new-choices) for an example. 
 
+## `createLabel`
```
Comment (djhi) [inline docs/SelectArrayInput.md:226]: :+1:

File: docs/AutocompleteArrayInput.md [hunk 9]
```diff
@@ -216,6 +216,37 @@ If you just need to ask users for a single string to create the new option, you
...
+```jsx
+<AutocompleteArrayInput
+    source="roles"
+    choices={choices}
+    create={<CreateRole />}
+    createLabel="Start typing to create a new item"
+/>
+```
+
+## `createItemLabel`
```
Comment (erwanMarmelab) [inline docs/AutocompleteArrayInput.md:233]: **Issue**: Could you please add a screenshot
```suggestion
![Create Label](./img/AutocompleteArrayInput-createLabel.png)

## `createItemLabel`
```

**Review by djhi — APPROVED** [review]
Excellent!

File: docs/AutocompleteArrayInput.md [hunk 10]
```diff
@@ -216,6 +216,37 @@ If you just need to ask users for a single string to create the new option, you
...
+
+If you set the `create` or `onCreate` prop, `<AutocompleteArrayInput>` lets users create new options. When the text entered by the user doesn't match any option, the input renders a "Create XXX" menu item at the bottom of the list.
```
Comment (erwanMarmelab) [inline docs/AutocompleteArrayInput.md:235]: **nitpick (non-blocking):** ‏
```suggestion
When you set the `create` or `onCreate` prop, `<AutocompleteArrayInput>` lets users create new options. When the text entered by the user doesn't match any option, the input renders a "Create XXX" menu item at the bottom of the list.
```

File: docs/AutocompleteInput.md [hunk 11]
```diff
@@ -60,8 +60,8 @@ The form value for the source must be the selected value, e.g.
...
+| `createItemLabel`          | Optional | `string`              | `ra.action .create_item`                                            | The label for the menu item allowing users to create a new choice. Used when the filter is not empty.                                                                                                                |
 | `debounce`                 | Optional | `number`              | `250`                                                               | The delay to wait before calling the setFilter function injected when used in a ReferenceInput.                                                                                                                     |
 | `emptyText`                | Optional | `string`              | `''`                                                                | The text to use for the empty element                                                                                                                                                                               |
```
Comment (erwanMarmelab) [inline docs/AutocompleteInput.md:66]: **todo:** ‏same here

File: docs/AutocompleteArrayInput.md [hunk 12]
```diff
@@ -216,6 +216,37 @@ If you just need to ask users for a single string to create the new option, you
...
+
+```jsx
+<AutocompleteArrayInput
+    source="roles"
+    choices={choices}
+    create={<CreateRole />}
+    createItemLabel="Add a new role: %{item}"
+/>
+```
+
```
Comment (erwanMarmelab) [inline docs/AutocompleteArrayInput.md:249]: **Issue**: Could you please add a screenshot
```suggestion

![Create Item Label](./img/AutocompleteArrayInput-createItemLabel.png)

```

File: docs/SelectArrayInput.md [hunk 13]
```diff
@@ -223,6 +223,21 @@ If you just need to ask users for a single string to create the new option, you
...
+
+```jsx
+<SelectArrayInput
+    source="roles"
+    choices={choices}
+    create={<CreateRole />}
+    createLabel="Add a new role"
+/>
+```
+
```
Comment (erwanMarmelab) [inline docs/SelectArrayInput.md:240]: **issue**: could you add a screenshot here too
